src/main/java/cam72cam/immersiverailroading/util/DataBlock.java

```diff
@@ -0,0 +1,207 @@
+package cam72cam.immersiverailroading.util;
+
+import cam72cam.immersiverailroading.ImmersiveRailroading;
+import cam72cam.mod.resource.Identifier;
+import com.google.gson.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.*;
+
+@SuppressWarnings("unused")
+public interface DataBlock {
+    Map<String, Value> getValueMap();
+    Map<String, List<Value>> getValuesMap();
+    Map<String, DataBlock> getBlockMap();
+    Map<String, List<DataBlock>> getBlocksMap();
+
+    default DataBlock getBlock(String key) {
+        return getBlockMap().get(key);
+    }
+
+    default List<DataBlock> getBlocks(String key) {
+        return getBlocksMap().get(key);
+    }
+
+    default Value getValue(String key) {
+        return getValueMap().getOrDefault(key, new Value() {
+            @Override
+            public Boolean asBoolean() {
+                return null;
+            }
+
+            @Override
+            public Integer asInteger() {
+                return null;
+            }
+
+            @Override
+            public Float asFloat() {
+                return null;
+            }
+
+            @Override
+            public Double asDouble() {
+                return null;
+            }
+
+            @Override
+            public String asString() {
+                return null;
+            }
+        });
```

Comment:
This is a fair bit strange in Java. I'd recommend having a static "emptyValue" sentinel that's reused, rather than using an anonymous class like this just for null-safety.

Reply:
Yeah, I'll probably move all the JSON related stuff into a helper class where I can do that instead of hacking it into an interface.